### PR TITLE
coprocessor: support builtin aggregation function `bit_and`, `'bit_or`, `bit_xor`.

### PR DESCRIPTION
**Pull request by spongedu**
The related modification in `tipb` is : https://github.com/pingcap/tipb/pull/67

### CODE REVIEW

**Comment by AndreMouche** [comment]
Please fix the CI

**Comment by AndreMouche** [comment]
@breeswish @winkyao PTAL

**Comment by spongedu** [comment]
@AndreMouche to make ci work,  https://github.com/pingcap/tipb/pull/67 must be checked in first.

**Comment by AndreMouche** [comment]
@hanfei1991 PTAL

**Comment by spongedu** [comment]
Should be merge after #2503 because some other `tipb` changes works with it.

**Comment by spongedu** [comment]
@AndreMouche @hanfei1991 @winkyao  PTAL

File: Cargo.lock [hunk 1]
```diff
@@ -1207,7 +1207,7 @@ dependencies = [
 "checksum thread-id 2.0.0 (registry+https://github.com/rust-lang/crates.io-index)" = "a9539db560102d1cef46b8b78ce737ff0bb64e7e18d35b2a5688f7d097d0ff03"
 "checksum thread_local 0.2.7 (registry+https://github.com/rust-lang/crates.io-index)" = "8576dbbfcaef9641452d5cf0df9b0e7eeab7694956dd33bb61515fb8f18cfdd5"
 "checksum time 0.1.38 (registry+https://github.com/rust-lang/crates.io-index)" = "d5d788d3aa77bc0ef3e9621256885555368b47bd495c13dd2e7413c89f845520"
-"checksum tipb 0.0.1 (git+https://github.com/pingcap/tipb.git)" = "<none>"
+"checksum tipb 0.0.1 (git+https://github.com/spongedu/tipb.git)" = "<none>"
```
[the remark this answers is not in the document]
Comment (spongedu) [reply Cargo.lock]: @AndreMouche  the `tipb` part has done in https://github.com/pingcap/tipb/pull/66 with some other modifications. Thus this pr should be done after  #2503 . I'll correct the cargo dependency after #2503 's checkin. The modification here is just to make tests happy.

**Comment by spongedu** [comment]
@AndreMouche I'm also working on the tidb part. It's in progress.

**Comment by AndreMouche** [comment]
Thanks, I've got it. We would like to review this PR after related PRs merged.

**Comment by spongedu** [comment]
@AndreMouche  the related PRs have been merged. I've revert the changes in `Cargo.lock` to fix dependency for `tipb`. PTAL.
BTW: `make test` and `make` works well on my laptop, but `ci/circleci` fails when build rocksDB. Do you know the reason?

**Comment by hanfei1991** [comment]
`coprocessor/select` is not used for tidb.....

Please implement it in `coprocessor/dag`, haha...

**Comment by AndreMouche** [comment]
@hanfei1991 @breeswish PATL

File: tests/coprocessor/test_select.rs [hunk 2]
```diff
@@ -468,10 +468,23 @@ impl<'a> Select<'a> {
...
+
+    fn bit_xor(self, col: Column) -> Select<'a> {
+        self.aggr_col(col, ExprType::Agg_BitXor)
+    }
+
     fn max(self, col: Column) -> Select<'a> {
         self.aggr_col(col, ExprType::Max)
     }
 
+
```
Comment (AndreMouche) [inline tests/coprocessor/test_select.rs]: Please remove the redundant empty  line

Comment (spongedu) [reply tests/coprocessor/test_select.rs]: ok

**Comment by AndreMouche** [comment]
Please fix the conflicts @spongedu

**Pull request merged**